resources/ts/Components/Integrations/Detail/UrlList.tsx

```diff
@@ -86,71 +120,108 @@ export const UrlList = ({
                       />
-                      {type !== IntegrationUrlType.Login && (
-                        <ButtonIcon
-                          icon={faTrash}
-                          onClick={() => setToBeDeletedId(url.id)}
-                          className="text-icon-gray"
-                        />
-                      )}
                     </div>
                   }
                 />
               ))}
-              {option.visible && (
-                <div className="flex flex-col">
-                  <FormElement
-                    label={t("details.integration_settings.new")}
-                    className=""
-                    component={
-                      <Input
-                        type="text"
-                        name="newUrl"
-                        className="md:min-w-[40rem]"
-                        onBlur={(e) =>
-                          onChangeNewUrl({
-                            environment: option.env,
-                            url: e.target.value,
-                            type,
-                          })
+              <div className="flex flex-col gap-2">
+                {newUrlFields.map(
+                  (field) =>
+                    field.startsWith(option.env) && (
+                      <FormElement
+                        key={field}
+                        error={errors[`newIntegrationUrls.${`${field}`}.url`]}
+                        component={
+                          <div className="flex gap-2">
+                            <Input
+                              type="text"
+                              name="newUrl"
+                              className="md:min-w-[40rem]"
+                              onChange={(e) =>
+                                onChangeNewUrl({
+                                  environment: option.env,
+                                  url: e.target.value,
+                                  type,
+                                  id: field,
+                                })
+                              }
+                            />
+                            <ButtonIcon
+                              icon={faTrash}
+                              onClick={() => {
+                                setToBeDeletedField(field);
+                                setIsDialogVisible(true);
+                              }}
+                              className="text-icon-gray"
+                            />
+                          </div>
                         }
                       />
-                    }
-                  />
-                </div>
-              )}
+                    )
+                )}
+              </div>
               {type !== IntegrationUrlType.Login && (
                 <ButtonSecondary
                   onClick={() => {
-                    option.changeVisibility(true);
+                    option.addNewUrlField(`${option.env + randomNumber}`);
```

Comment:
What is the reason of the `randomNumber`?

Reply:
The PR isn't ready for review yet, my bad.

This is remaining code I still have to refactor